### PR TITLE
Misc log-related fixes

### PR DESCRIPTION
**Pull request by eriknordmark**
Separated out from the larger #1391

### CODE REVIEW

File: pkg/pillar/base/logobjecttypes.go [hunk 1]
```diff
@@ -172,13 +176,28 @@ func InitLogObject(logBase *LogObject, object *LogObject, objType LogObjectType,
...
+			value)
+	}
+
+	object = new(LogObject)
 	object.Initialized = true
 	fields := make(map[string]interface{})
 	fields["source"] = agentName
 	fields["pid"] = agentPid
 	object.Fields = fields
+	logSourceObjectMap.Store(agentName, object)
```
Comment (gkodali-zededa) [inline pkg/pillar/base/logobjecttypes.go:200]: I suppose the assumption is that these source log objects once created will only be destroyed along with zedbox, right?

**Review by gkodali-zededa — APPROVED** [review]
LGTM

File: pkg/pillar/base/logobjecttypes.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/base/logobjecttypes.go:200]: Yes, because there is currently no way to shut down a zedbox microservice.

**Review by rvs — COMMENTED** [review]
LGTM

**Pull request merged**